src/main/java/org/apache/sysml/api/python/SystemML.py

```diff
@@ -247,3 +259,161 @@ def getStringRDD(self, varName, format):
+            self.tol = tol
+            if C <= 0:
+                raise Exception('C has to be positive')
+            reg = 1.0 / C
+            self.reg = reg
+            self.updateLog()
+            if solver != 'newton-cg':
+                raise Exception('Only newton-cg solver supported')
+             
+        def updateLog(self):
+            self.log.setMaxOuterIter(self.max_iter)
+            self.log.setMaxInnerIter(self.max_inner_iter) 
+            self.log.setRegParam(self.reg)
+            self.log.setTol(self.tol)
+            self.log.setIcpt(self.icpt)
+            
+        def convertToPandasDF(self, X):
+            if isinstance(X, np.ndarray):
+                colNames = []
+                numCols = getNumCols(X)
+                for i in range(0, numCols):
+                    colNames = colNames + [ str('C' + str(i))]
+                pdfX = pd.DataFrame(X, columns=colNames)
+            elif isinstance(X, pd.core.frame.DataFrame):
+                pdfX = X
+            else:
+                raise Exception('The input type not supported')
+            return pdfX
+            
+        def tolist(self, inputCols):
+            if isinstance(inputCols, pd.indexes.base.Index):
+                return inputCols.get_values().tolist()
+            elif isinstance(inputCols, list):
+                return inputCols
+            else:
+                raise Exception('inputCols should be of type pandas.indexes.base.Index or list')
+                
+        def assemble(self, pdf, inputCols, outputCol):
+            tmpDF = self.sqlCtx.createDataFrame(pdf, self.tolist(pdf.columns))
+            assembler = VectorAssembler(inputCols=self.tolist(inputCols), outputCol=outputCol)
+            return assembler.transform(tmpDF)
+            
+        def _fit(self, X):
+            if hasattr(X, '_jdf') and 'features' in X.columns and 'label' in X.columns:
+                self.model = self.log.fit(X._jdf)
+                return self
+            else:
+                raise Exception('Incorrect usage: Expected dataframe as input with features/label as columns')
+            
+        # TOOD: Ignoring kwargs
+        def fit(self, X, *args, **kwargs):
+            self.updateLog()
+            numArgs = len(args) + 1
+            if numArgs == 1:
+                return self._fit(X)
+            elif numArgs == 2 and (isinstance(X, np.ndarray) or isinstance(X, pd.core.frame.DataFrame)):
+                y = args[0]
+                if self.transferUsingDF:
+                    pdfX = self.convertToPandasDF(X)
+                    pdfY = self.convertToPandasDF(y)
+                    if getNumCols(pdfY) != 1:
+                        raise Exception('y should be a column vector')
+                    if pdfX.shape[0] != pdfY.shape[0]:
+                        raise Exception('Number of rows of X and y should match')
+                    colNames = pdfX.columns
+                    pdfX['label'] = pdfY[pdfY.columns[0]]
```

Comment:
The current behaviour explicitly renames the columns of `y` to label even if the user provides `y` as a DataFrame. Are we okay with that?